web/src/components/analyze/AnalyzeBase.pug

```diff
@@ -0,0 +1,17 @@
+v-layout(row, fill-height)
+  block toolbar
```

Comment:
I was still conflicted about this, so I chatted with some more experienced Vue developers.  I provided [this](https://medium.com/js-dojo/extending-vuejs-components-42fefefc688b) as the example, because I think it accurately summarized this pattern.   I'm just going to reproduce the discussion here so I don't add any bias to my summary.

----------------------

## subdavisLast Friday at 8:09 AM

Any Vue veterans have thoughts about this inheritance pattern:

“Extending VueJS Components” by Anthony Gore https://link.medium.com/YMsFhSFSWZ

My instinct is to consider this a bit of an anti-pattern and to instead try to solve the problem with scoped slots, but I want a few more data points.  Ty!
 
## jack Last Friday at 11:29 AM

Hehe, ben co-runs the best workshop on this question.

I'd definitely agree, that I've found composition through slots to be more maintainable & fit with the goals of a UI better.

Can also attest to the frustration, in large JS Vue projects, of trying to click through to find where something is originally defined, whether that's a mixin or a base component being extended. Since the merge strategy is a Vue-only thing, there's no way for an IDE (or TypeScript) to help you go find "yup, here's where this.myBaseMethod is defined"
Inheritance from a base component is probably most OK if you're never going to try to inherit more than 1 level; any more than 1 level leads to a lot of pain later.
 
## subdavisLast Friday at 11:59 AM

Thanks jack, I appreciate the response :)
 
## ben Friday at 2:18 PM

Agreed with jack on this one. Composition is really the name of the game over inheritance, hence the Composition API coming up in Vue 3
Like it's good to know this technique exists, but I'm especially not a fan of using pug to extend a base template

---------

I think that's a lot of caveat to accept when adopting a pattern.  IMO, it might be good to avoid doing this just for the sake of style consistency and so that some future developer doesn't see it and continue down the path.  Also, once we are ready to adopt Vue 3, it will likely be easier to migrate if we stick to common patterns.

Reply:
take a look at https://github.com/girder/metabulo/pull/191 in which I already migrated to a wrapper component and mixin style

Reply:
Independent of the question of composition vs. inheritance, I think using Pug templating features inside Vue is a bad idea in general.

For this specific use case, mixins and scoped slots are code reuse patterns provided by Vue.  We should default to those unless there is a very clear reason not to.